Gene profiling techniques such as DNA microarrays and RNA sequencing are important microbiological tools. Unfortunately their efficacy is often jeopardized by heterogeneous samples. Practitioners often compare gene expressions measurements across samples. The difference in expressions across samples will be due in part to biological differences in similar cells and in part due to differences in the types of cells comprising each sample. Consequently the differences in sample composition confounds expression changes since we can't distinguish differences in expression coming from biological differences in similar cells and changes in expression arising from different composition of the cell populations in the samples. In order to separate the effects of sample heterogeneity from other biological factors methods of cell type deconvolution have been developed. In this paper we review the literature on such cell type deconvolution algorithms and propose new deconvolution methodology. Our algorithm differs from the prevailing modeling of expressions in heterogeneous samples as convex combinations of expressions in pure samples. Instead we directly model the interaction between the quantity of RNA in a sample and expression measurements in the sample. Using real data we demonstrate that our algorithm has a competitive prediction error with existing methods even though it is much simpler. Finally we discuss the next paths we wish to explore in this line of research. 
